Fix JavaScriptCore API headers to work with the Clang Module Verifier.

MenloDorian opened the pull request:
#### fc2e779ba915c68637aa2b57e0ab38545c0d299a
<pre>
Fix JavaScriptCore API headers to work with the Clang Module Verifier.
<a href="https://bugs.webkit.org/show_bug.cgi?id=279072">https://bugs.webkit.org/show_bug.cgi?id=279072</a>
<a href="https://rdar.apple.com/131335204">rdar://131335204</a>

Reviewed by Elliott Williams, Tim Horton, and Keith Miller.

Changes requires by the Clang Module Verifier:
1. Removed [extern_c] from the module map declaration.
2. Fixed the top level umbrella JavaScriptCore.h to not contain any conditional includes.

Additional changes:
3. As a consequence of (2), changed #import&apos;s to #include&apos;s and added header guards to all the relevant files.
4. As a clean up, changed use of C++ style // comments to C style /* */ comments.
5. To prevent future regressions, added running the Clang Module Verifier to JavaScriptCore.xcconfig.

* Source/JavaScriptCore/API/JSContext.h:
* Source/JavaScriptCore/API/JSExport.h:
* Source/JavaScriptCore/API/JSManagedValue.h:
* Source/JavaScriptCore/API/JSValue.h:
* Source/JavaScriptCore/API/JSVirtualMachine.h:
* Source/JavaScriptCore/API/JavaScriptCore.h:
* Source/JavaScriptCore/Configurations/JavaScriptCore.xcconfig:
* Source/JavaScriptCore/JavaScriptCore.modulemap:

Canonical link: <a href="https://commits.webkit.org/283128@main">https://commits.webkit.org/283128@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/333f2d61402dd8b97b20b880118aa5a1928e68c1

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [❌ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/65247 "6 style errors") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/44616 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/17861 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/69271 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/15853 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/52398 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/16135 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/52414 "Passed tests") | [✅ 🧪 wincairo-tests](https://ews-build.webkit.org/#/builders/60/builds/10979 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/68313 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/41236 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/56475 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/33037 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/37907 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/13846 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/14730 "Built successfully") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/58360 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/59789 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/14186 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/70976 "Built successfully") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/64490 "Built successfully and passed tests") | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/87/builds/9199 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/13664 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/59740 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/86/builds/9231 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/56536 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/60014 "Passed tests") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/88/builds/7600 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/1273 "Passed tests") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/86258 "Built successfully") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/9907 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/40426 "Built successfully") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/15197 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/41503 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/42684 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/41247 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

ian-twilightcoder commented on Source/JavaScriptCore/Configurations/JavaScriptCore.xcconfig:
```diff
@@ -24,6 +24,11 @@
 #include "BaseTarget.xcconfig"
 
 DEFINES_MODULE = YES;
+ENABLE_MODULE_VERIFIER = $(USE_INTERNAL_SDK);
```
I think you can just make this YES. Where it's not supported in the public SDK, the setting is ignored.

MenloDorian replied:
I'll leave this as is because it documents intent.

ian-twilightcoder commented on Source/JavaScriptCore/Configurations/JavaScriptCore.xcconfig:
```diff
@@ -24,6 +24,11 @@
+ENABLE_MODULE_VERIFIER = $(USE_INTERNAL_SDK);
+OTHER_MODULE_VERIFIER_FLAGS = $(OTHER_MODULE_VERIFIER_FLAGS_$(WK_EMPTY_$(WK_OVERRIDE_FRAMEWORKS_DIR)));
+OTHER_MODULE_VERIFIER_FLAGS_ = -- -iframework $(WK_PRIVATE_SDK_DIR)$(WK_OVERRIDE_FRAMEWORKS_DIR) -isystem $(WK_PRIVATE_SDK_DIR)$(WK_ALTERNATE_WEBKIT_SDK_PATH)$(WK_LIBRARY_HEADERS_FOLDER_PATH);
+MODULE_VERIFIER_VERBOSE = NO;
```
NO is the default, don't need to specify this

kmiller68 replied:
I would actually leave it specified since it makes it easier to find/enable when debugging in the future.

MenloDorian replied:
Since it doesn't cost much to have it, I'll leave this here to make it clear to other folks trying to debug failures in this area in the future (but are not familiar with this as I) that this helpful option is available.

emw-apple commented on Source/JavaScriptCore/API/JavaScriptCore.h:
```diff
@@ -29,14 +29,10 @@
-#import <JavaScriptCore/JSExport.h>
-
-#endif
+#include <JavaScriptCore/JSContext.h>
+#include <JavaScriptCore/JSValue.h>
+#include <JavaScriptCore/JSManagedValue.h>
+#include <JavaScriptCore/JSVirtualMachine.h>
+#include <JavaScriptCore/JSExport.h>
```
Is the order important here? Style checker is complaining about sort order.

MenloDorian replied:
I don't know if the order is important, but given that this is an API file, I rather not change the order.

Commit:
Fix JavaScriptCore API headers to work with the Clang Module Verifier.
https://bugs.webkit.org/show_bug.cgi?id=279072
rdar://131335204

Reviewed by Elliott Williams, Tim Horton, and Keith Miller.

Changes requires by the Clang Module Verifier:
1. Removed [extern_c] from the module map declaration.
2. Fixed the top level umbrella JavaScriptCore.h to not contain any conditional includes.

Additional changes:
3. As a consequence of (2), changed #import's to #include's and added header guards to all the relevant files.
4. As a clean up, changed use of C++ style // comments to C style /* */ comments.
5. To prevent future regressions, added running the Clang Module Verifier to JavaScriptCore.xcconfig.

* Source/JavaScriptCore/API/JSContext.h:
* Source/JavaScriptCore/API/JSExport.h:
* Source/JavaScriptCore/API/JSManagedValue.h:
* Source/JavaScriptCore/API/JSValue.h:
* Source/JavaScriptCore/API/JSVirtualMachine.h:
* Source/JavaScriptCore/API/JavaScriptCore.h:
* Source/JavaScriptCore/Configurations/JavaScriptCore.xcconfig:
* Source/JavaScriptCore/JavaScriptCore.modulemap:

Canonical link: https://commits.webkit.org/283128@main
